core/array.cpp

```diff
@@ -417,45 +417,37 @@ Variant Array::pop_front() {
 
 Variant Array::min() const {
 
-	Variant minval;
-	for (int i = 0; i < size(); i++) {
-		if (i == 0) {
-			minval = get(i);
-		} else {
-			bool valid;
-			Variant ret;
-			Variant test = get(i);
-			Variant::evaluate(Variant::OP_LESS, test, minval, ret, valid);
-			if (!valid) {
-				return Variant(); //not a valid comparison
-			}
-			if (bool(ret)) {
-				//is less
-				minval = test;
-			}
+	Variant minval = get(0);
+	for (int i = 0; i < size(); ++i) {
+		bool valid;
+		Variant ret;
+		Variant test = get(i);
+		Variant::evaluate(Variant::OP_LESS, test, minval, ret, valid);
```

Comment:
If `evaluate` is an expensive function, this PR could actually worsen performance since you are calling it one more time than than previously. To match previous behaviour you probably want to start the loop at index 1.

Also, before this change if this function was called on an empty array, an empty Variant would be returned. After this change that call would cause a crash.

Have you profiled the effect of this change? Getting the first value outside the loop is a bit cleaner in my opinion, but I don't think it should be labeled as an optimisation until this code path can shown to take a significant amount of time when profiling, and if so, that this change fixes that. In my experience if statements and pre/post integer increments never show up in profiling results, even if you're doing 1000 too many of them per frame they're still orders of magnitude faster than other operations like I/O or physics steps.

(Same comments for the changes to `max`)

Reply:
Additionally, why not start from `int x = 1`?
```
Variant minval = get(0);
	for (int i = 0; i < size(); ++i) {
```

Reply:
An argument could be made that if statements inside a loop is bad? Because then its not trivial to unroll it.

Reply:
@ajweeks Could it be that you confused what is changed here? If you tap on cogwheel and set layout to "Split", left part of  diff is old code, and right side is new code.